crates/primitives/src/env.rs

```diff
@@ -318,6 +446,11 @@ impl Env {
             return Err(InvalidTransaction::AccessListNotSupported);
         }
 
+        // Check that the max fee per blob gas is not set for transactions before CANCUN
+        if !SPEC::enabled(SpecId::CANCUN) && self.tx.max_fee_per_blob_gas.is_some() {
+            return Err(InvalidTransaction::MaxFeePerBlobGasNotSupported);
+        }
+
```

Comment:
Add additional check for `blob_hashes.is_empty()` before CANCUN

Reply:
I can also remove `max_fee_per_blob_gas` since I don't think we'll be using it in REVM?